foursight_core/react/api/envs.py

```diff
@@ -0,0 +1,82 @@
+import copy
+from typing import Optional
+import os
+from dcicutils import ff_utils
+
+
+class Envs:
+
+    def __init__(self, known_envs: list):
+        # This known_envs should be the list of annotated environment name objects
+        # as returned by app_utils.get_unique_annotated_environment_names, where each
+        # object contains these fields: name, short_name, full_name, public_name, foursight_name
+        self.known_envs = known_envs
+
+    def get_known_envs(self) -> list:
+        return self.known_envs
+
+    def get_known_envs_with_gac_names(self, gac) -> list:
```

Comment:
So if it's adding information, this is a legit reason to have these be dictionaries and not names. (I mentioned maybe not having dictionaries somewhere, so you can ignore that.) But I think the only dictionary keys it should need are something to hold the foursight_env_name and something to hold the gac_name. The others aren't needed if you use the env_utils functions well.

The whole purpose of a `foursight_env_name` is to have a single canonical name so you can do fewer compares, not to add a fourth thing you have to compare to every time.

Reply:
I get that now, but for now gonna leave it just because I'd have to rework a bit of the UI code as it looks at these various name variants (sanity checking and whatnot); adding TODO comments.